target/bin/listmailuser

```diff
@@ -90,15 +90,9 @@ function _alias_list_for_account
   local MAIL_ACCOUNT=${1}
 
   # `__db_list_already_contains_value` would be a more reliable check:
-  function _account_has_an_alias
-  {
-    local ANY_ALIAS='\S\+\s'
-    grep -qis "^${ANY_ALIAS}.*${MAIL_ACCOUNT}" "${DATABASE_VIRTUAL}"
-  }
```

Comment:
This check you've removed ensured that the alias is defined at the start of the line, and that the target mail account comes after some white-space (`ANY_ALIAS`), as there may be more than one account expanded from the alias.

Your change insists that the mail account is at the end of the line. This is not always the case when the alias has more than one account (_or other valid values such as another alias_) to expand to. Your fix introduces a new bug :(

The proper fix would be to additionally ensure `MAIL_ACCOUNT` has white-space surrounding it as well (unless it's the last value on the line.

The comment right above this method (_that you retained, despite removing the method_) points out a more thorough check, in the event this kind of problem happened :sweat_smile: 

https://github.com/docker-mailserver/docker-mailserver/blob/5f671a4c0b0e41e82cc6432d011a342b9e881094/target/scripts/helpers/database/db.sh#L130-L140

Which notes it's an internal method, so will either need refactoring or the easier way is to maintain similar code instead of extracting out a common utility that can be shared elsewhere?

https://github.com/docker-mailserver/docker-mailserver/blob/5f671a4c0b0e41e82cc6432d011a342b9e881094/target/scripts/helpers/database/db.sh#L38-L66

---

That said, this logic was still fine AFAIK, it only checked if the alias existed on a line for a given account in our alias DB / config file.

So perhaps revert the change and keep the condition? Just fix the issue with the actual filtering logic below?